datalad_next/iterable_subprocess/test_iterable_subprocess.py

```diff
@@ -332,9 +332,9 @@ def test_error_returncode_available_from_generator_with_exception():
     assert ls.returncode != 0
 
 
-def test_success_returncode_available_from_generator_with_exception():
+def test_returncode_available_from_generator_with_exception():
     with pytest.raises(StopIteration):
         with iterable_subprocess(['echo', 'a'], ()) as echo:
             while True:
                 next(echo)
-    assert echo.returncode == 0
+    assert echo.returncode in (0, -15)
```

Comment:
```suggestion
    # On a Linux system, all exceptions that are raised before the subprocess
    # exited will lead to a -15 return code. If StopIteration is raised, the
    # subprocess will either have terminated which results in a 0-return code,
    # or the subprocess is still running and will therefore be terminated which
    # results in a -15 return code. Any other exception than StopIteration,
    # e.g. a CommandError because echo could not be found, would lead to an
    # early test-exit and not proceed to the assign-statement.
    assert echo.returncode in (0, -15)
```

Let's import the rationale for the setup from the PR comment into the code.

Reply:
Cool thanks.